packages/cinterop/src/darwin/kotlin/io/realm/internal/interop/RealmInterop.kt

```diff
@@ -1101,6 +1101,34 @@ actual object RealmInterop {
         return AuthProvider.of(realm_wrapper.realm_auth_credentials_get_provider(credentials.cptr()))
     }
 
+    actual fun realm_app_email_password_provider_client_register_email(
+        app: NativePointer,
+        email: String,
+        password: String,
+        callback: CinteropVoidCallback
+    ) {
+        memScoped {
+            realm_wrapper.realm_app_email_password_provider_client_register_email(
+                app.cptr(),
+                email,
+                password.toRString(this),
+                staticCFunction { userData, error ->
+                    val userDataCallback = safeUserData<CinteropVoidCallback>(userData)
```

Comment:
Even though the code is more compact than for JVM we should probably abstract this _callback handling_ into a method too.